pages/a/1k1-azure-post.html

```diff
@@ -0,0 +1,239 @@
+trigger a complex cluster quickly.
+
+An ARM template has a simple structure:
+
+```bash
+{
+   "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+   "contentVersion": "",
+   "parameters": {  },
+   "variables": {  },
+   "resources": [  ],
+   "outputs": {  }
+}
+```
+
+It is worth highlighting the resource dependency management in ARM templates, which provide a
+cohesive template structure and reliable deployment of resources.
+
+
+### Deployment Setup
+
+Taking into account limitations of the Azure platform, especially number of resources per single
+resource group and number of virtual machines in single availability set and convenient way for
+possible redeploying of resources, we came to a conclusion to split the resource setup into logical
+self contained units defined by their function:
+
+    * 1 Network infrastructure group
+    * 1 Docker Swarm managers group
+    * 1 Crate masters group
+    * 11 Scale unit groups
+    * 1 Ganglia monitoring group
+
+#### Network Infrastructure
+
+As a fundamental component of the overall deployment architecture, the resource group units serve the
+[network resource](https://github.com/crate/p-azure/blob/master/arm-templates/network.deploy.json)
+group. The network ARM template defines only three types of resources to be deployed, they are:
+
+    * Virtual Network
+    * Network Security Group
+    * Public IP addresses used for Swarm Manager VMs
+
+It is also possible to reference resources by ID from other groups by providing its type and an
+explicit path to it:
+
+```bash
+resourceId ([resourceGroupName], resourceType, resourceName1, [resourceName2]...)
+```
+
+For instance, we reference the virtual network from the network resource group in each of ARM
+template as following:
+
+```bash
+    ...
+    "virtualNetworkName": "vnet1k1",
+    "resourceGroupNetworkName": "1k1network",
+    ...
+    "vnetID": "[resourceId(variables('resourceGroupNetworkName'), 'Microsoft.Network/virtualNetworks'
+        , variables('virtualNetworkName'))]",
+    ...
+```
+
+#### Docker Swarm managers
+
+Docker Swarm managers and Consul agents run in the dedicated [Docker Swarm managers] group
+(https://github.com/crate/p-azure/blob/master/arm-templates/docker/swarmmanagers.json).
+
+    * 3 Network Interfaces
+    * 3 Virtual Machine
+    * 6 Microsoft Azure extensions
+    * 1 Availability set
+    * 1 Storage account
+
+#### Crate nodes
+
+All in all, to have 1001 Crate data nodes in the cluster,
+[11 scale unit](https://github.com/crate/p-azure/blob/master/arm-templates/docker/scaleunit.json)
+groups must be deployed. With an idea of keeping the template as simple as possible, while
+satisfying existing constraints, the following structure of the template was proposed:
+
+    * 100 Network Interfaces
+    * 100 Virtual Machine
+    * 200 Microsoft Azure extensions
+    * 1 Availability set
+    * 5 Storage accounts
+
```

Comment:
Maybe add a reason like: "So we were able to add one scaleunit after the other."